packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { calleesFlamebearer } from './sandwichViewProfiles';
+import { flamebearersToTree } from './flamebearersToTree';
+
+import { tree1 } from './testData';
+
+jest.mock('./flamebearersToTree', () => ({
+  flamebearersToTree: jest.fn(),
+}));
+
+describe('Sandwich view profiles', () => {
+  beforeAll(() => {
+    (flamebearersToTree as jest.Mock).mockReturnValue(tree1);
+  });
+
+  it('should return correct callees flamebearer (single target function name appearance)', () => {
+    expect(calleesFlamebearer({} as any, 'name-2-2')).toMatchObject({
+      format: 'single',
+      levels: [
+        [0, 400, 0, 0],
+        [0, 400, 0, 1],
+        [0, 400, 100, 2],
+        [0, 300, 0, 3],
+        [0, 150, 150, 4, 150, 150, 150, 5],
+      ],
+      maxSelf: 150,
+      names: [
+        'total',
+        'name-2-2',
+        'name-3-1',
+        'specific-function-name',
+        'name-5-1',
+        'name-5-2',
+      ],
+      numTicks: 400,
+      sampleRate: 100,
+    });
```

Comment:
I think this is a bad idea. With a test i should able to inspect its intent by reading the code/artifacts. Here the `flamebearer` format is for all effects, gibberish.

Can you convert it back to another representation? I suggest the `tree` one which is what most people are used to.

Reply:
So what I am suggesting is:

Create a tree by hand.
Convert to flamebearer using one our functions.
Run calleesFlamebearer
Convert its output to a tree
Compare if the output tree is the same as EXPECTED_TREE (something you create manually)

This should be a test made of pure functions (input -> output), no need to mock anything.

Reply:
I don't think it's a "bad" idea -- curious why you did choose this format instead of a tree @dogfrogfog ? Did you consider the tree too?

Reply:
so i chose flamebearer because this is the function's return value...but ye I think the way Eduardo suggested is better and easier to understand... I will refactor this part